dapp/src/components/buySell/ApproveSwap.js

```diff
@@ -99,27 +114,14 @@ const ApproveSwap = ({
     stableCoinToApprove,
     isMobile,
   }) => {
-    const capitalized = selectedSwap.name.charAt(0).toUpperCase()
-    const noncapitalized =
-      selectedSwap.name === 'uniswapV2'
-        ? selectedSwap.name.slice(1, 7)
-        : selectedSwap.name.slice(1)
-    const origin =
-      selectedSwap.name === 'flipper' || selectedSwap.name === 'vault'
-        ? 'the '
-        : ''
-    const vault = selectedSwap.name === 'vault' ? 'Origin ' : ''
-    const coin = stableCoinToApprove.toUpperCase()
-    const route = `${origin} ${vault} ${capitalized}${noncapitalized} to use your ${coin}`
-    const routeMobile = `${origin} ${vault} ${capitalized}${noncapitalized}`
     if (stage === 'waiting-user') {
       return fbt(
         'Waiting for you to confirm...',
         'Waiting for you to confirm...'
       )
     }
     if (stage === 'waiting-network') {
-      const waitingNetworkMessage = `${origin} ${vault} ${capitalized}${noncapitalized}`
+      const waitingNetworkMessage = routeConfig[selectedSwap.name].name(false)
```

Comment:
this bit should be easier to understand to somebody that has not written the code. Or to the person who wrote it and need to revisit it in 6 months.